src/planner/binder/query_node/bind_macro_select_node.cpp

```diff
@@ -0,0 +1,165 @@
+
+#include "duckdb/catalog/catalog.hpp"
+#include "duckdb/catalog/catalog_entry/macro_catalog_entry.hpp"
+#include "duckdb/parser/expression/function_expression.hpp"
+#include "duckdb/parser/expression/subquery_expression.hpp"
+#include "duckdb/parser/parsed_expression_iterator.hpp"
+#include "duckdb/planner/expression_binder.hpp"
+#include "duckdb/common/string_util.hpp"
+
+#include "duckdb/common/limits.hpp"
+#include "duckdb/common/string_util.hpp"
+#include "duckdb/execution/expression_executor.hpp"
+#include "duckdb/main/config.hpp"
+#include "duckdb/parser/expression/columnref_expression.hpp"
+#include "duckdb/parser/expression/comparison_expression.hpp"
+#include "duckdb/parser/expression/constant_expression.hpp"
+#include "duckdb/parser/expression/subquery_expression.hpp"
+#include "duckdb/parser/expression/function_expression.hpp"
+#include "duckdb/parser/query_node/select_node.hpp"
+#include "duckdb/parser/tableref/joinref.hpp"
+
+#include "duckdb/catalog/catalog_entry/scalar_function_catalog_entry.hpp"
+#include "duckdb/planner/binder.hpp"
+
+namespace duckdb {
+
+// void ReplaceMacroSelectParametersRecursive(unique_ptr<ParsedExpression> &expr, MacroBinding *macro_binding);
+
+void ReplaceMacroSelectParametersRecursive(unique_ptr<ParsedExpression> &expr, MacroBinding *macro_binding) {
+	switch (expr->GetExpressionClass()) {
```

Comment:
This function looks duplicated from `ReplaceMacroParametersRecursive`, any reason not to use the old function?

Reply:
Yes am aware of this  
The original ReplaceMacroParametersRecursive()  is declared in ExpressionBinder scope  as 

~~~
void ExpressionBinder::ReplaceMacroParametersRecursive(unique_ptr<ParsedExpression> &expr)
~~~

And uses this->macro_binding to pass  paramerized  function expressions.

I am having trouble calling this function from the Binder:: scope with my new Macro Method.
Binder::BindMacroSelect(FunctionExpression(). Ive tried quite a few things.